<launch>

  <!-- settings -->
  <arg name="initial_selector_mode" default="local" description="local, remote" />

  <!-- service -->
  <arg name="service/select_external_command" default="~/select_external_command"/>

  <!-- input local -->
  <arg name="input/local/control_cmd" />
  <arg name="input/local/shift_cmd" />
  <arg name="input/local/turn_signal_cmd" />
  <arg name="input/local/heartbeat" />

  <!-- input remote -->
  <arg name="input/remote/control_cmd" />
  <arg name="input/remote/shift_cmd" />
  <arg name="input/remote/turn_signal_cmd" />
  <arg name="input/remote/heartbeat" />

  <!-- output -->
  <arg name="output/control_cmd" default="/external/selected/external_control_cmd"/>
  <arg name="output/shift_cmd" default="/external/selected/shift_cmd"/>
  <arg name="output/turn_signal_cmd" default="/external/selected/turn_signal_cmd"/>
  <arg name="output/heartbeat" default="/external/selected/heartbeat"/>
  <arg name="output/current_selector_mode" default="~/current_selector_mode"/>

  <!-- node -->
  <node pkg="external_cmd_selector" exec="external_cmd_selector" name="external_cmd_selector" output="screen">
    <remap from="~/service/select_external_command" to="$(var service/select_external_command)" />
    <remap from="~/input/local/control_cmd" to="$(var input/local/control_cmd)" />
    <remap from="~/input/local/shift_cmd" to="$(var input/local/shift_cmd)" />
    <remap from="~/input/local/turn_signal_cmd" to="$(var input/local/turn_signal_cmd)" />
    <remap from="~/input/local/heartbeat" to="$(var input/local/heartbeat)" />
    <remap from="~/input/remote/control_cmd" to="$(var input/remote/control_cmd)" />
    <remap from="~/input/remote/shift_cmd" to="$(var input/remote/shift_cmd)" />
    <remap from="~/input/remote/turn_signal_cmd" to="$(var input/remote/turn_signal_cmd)" />
    <remap from="~/input/remote/heartbeat" to="$(var input/remote/heartbeat)" />
    <remap from="~/output/control_cmd" to="$(var output/control_cmd)" />
    <remap from="~/output/shift_cmd" to="$(var output/shift_cmd)" />
    <remap from="~/output/turn_signal_cmd" to="$(var output/turn_signal_cmd)" />
    <remap from="~/output/heartbeat" to="$(var output/heartbeat)" />
    <remap from="~/output/current_selector_mode" to="$(var output/current_selector_mode)" />
    <param name="initial_selector_mode" value="$(var initial_selector_mode)" />
  </node>

</launch>
